TF_Settings_Web/src/Components/Pages/CameraPage.tsx

```diff
@@ -1,18 +1,240 @@
+        }));
+    }
+
+    roundToFiveDecimals(numberIn: number): number {
+        return Math.round(numberIn * 100000) / 100000;
+    }
+
+    componentDidUpdate(prevProps: {}, prevState: physicalState): void {
+        if (this.state !== null &&
+            this.state !== prevState) {
+            let physicalConfigState: PhysicalConfig = {
+                ...this.state.physicalConfig,
+                LeapPositionRelativeToScreenBottomM: {
+                    X: this.state.cameraLeftToRight / 100,
+                    Y: this.state.cameraHeight / 100,
+                    Z: -this.state.cameraDistanceFromScreen / 100
+                },
+                LeapRotationD: {
+                    ...this.state.physicalConfig.LeapRotationD,
+                    X: this.state.cameraRotation
+                },
+                ScreenHeightM: this.state.screenHeight / 100,
+                ScreenRotationD: this.state.screenTilt
+            }
+            ConfigurationManager.RequestConfigFileChange(null, physicalConfigState, this.configChangeCbHandler.bind(this));
+        }
+    }
+
+    configChangeCbHandler(result: WebSocketResponse): void {
+        if (result.status !== "Success") {
+            console.error(`Failed to set config state! Info: ${result.message}`);
+        }
+    }
+
+
+    onScreenHeightChanged(e: React.FormEvent<HTMLInputElement>): void {
+        const value = parseFloat(e.currentTarget?.value) ?? 0;
+        this.setState(() => ({
+            screenHeight: value
+        }));
+    }
+    onCameraHeightChanged(e: React.FormEvent<HTMLInputElement>): void {
+        const value = parseFloat(e.currentTarget?.value) ?? 0;
+        this.setState(() => ({
+            cameraHeight: value
+        }));
+    }
+    onCameraLeftToRightChanged(e: React.FormEvent<HTMLInputElement>): void {
+        const value = parseFloat(e.currentTarget?.value) ?? 0;
+        this.setState(() => ({
+            cameraLeftToRight: value
+        }));
+    }
+    onScreenTiltChanged(e: React.FormEvent<HTMLInputElement>): void {
+        const value = parseFloat(e.currentTarget?.value) ?? 0;
+        this.setState(() => ({
+            screenTilt: value
+        }));
+    }
+    onCameraRotationChanged(e: React.FormEvent<HTMLInputElement>): void {
+        const value = parseFloat(e.currentTarget?.value) ?? 0;
+        this.setState(() => ({
+            cameraRotation: value
+        }));
+    }
+    onCameraDistanceFromScreenChanged(e: React.FormEvent<HTMLInputElement>): void {
+        const value = parseFloat(e.currentTarget?.value) ?? 0;
+        this.setState(() => ({
+            cameraDistanceFromScreen: value
+        }));
+    }
 
-export class CameraPage extends Page {
-    private divStyle: CSSProperties = {
-        flex: '1 1 auto',
-        width: '100%',
-        height: '800px',
-        backgroundColor: 'red',
-    };
+    onScreenHeightClicked(): void {
+        this.setState({...this.state, selectedView: "screenHeight"});
```

Comment:
Two things here: 

1. there's some inconsistency in the use of an object and lambda inside this.setState to change the values, and I'm not sure there's a reason for it? Would be nice to be consistent if not.
2. We're using `this.state` inside of setState, but setState is executed asynchronously. We should probably be using the `state` parameter that gets passed to the lambda within `setState` to be as up-to-date as possible if we're modifying said state, as per [the docs here](https://reactjs.org/docs/react-component.html#setstate)

Reply:
I have now updated those as well